BaseTransaction.java

```
@@ -41,3 +41,13 @@ public void printTransactionDetails() {
     @Override
     public abstract void apply(BankAccount ba); // To be implemented by subclasses
 }
+
+@Override
+public void apply(BankAccount ba) {
```

You can also throw an exception in this class, and catch it in the client class